### PR TITLE
New/ni fake

### PR DESCRIPTION
**Pull request by marcoskirsch**
[X] This contribution adheres to [CONTRIBUTING.md]

### What does this Pull Request accomplish?

Adds a new fake MI driver "NI-FAKE" to the set of Python bindings generated by nimi-python. The goal is to use this fake driver for testing the code-generator and unit-testing the generated code. As such, NI-FAKE should contain one but no more than one example for every type of attribute/function and unit-tests that verify that the Python code is calling into the mock library correctly.

Neither the metadata nor the unit tests have been fully fleshed out.

### Why should this Pull Request be merged?

This is a refinement of our testing strategy. There's no point writing unit tests for the real bindings (such as NI-DMM, NI-SCOPE, etc.) since that wouldn't validate the metadata anyway (we're mocking the calls to the library). Validating the metadata requires system tests.
Also, by attempting to write unit tests for the individual bindings we will either risk lots of repetition which is bad or missing coverage which is worse.

By having a concise "for unit testing" driver that has one example for every thing the code-generator needs to handle, we can ensure that we have good unit-test coverage. By not having redundancy (i.e. multiple attributes of the same type) the use of code coverage tools becomes viable.

Once this is merged and things have been fleshed out, we can remove the unit-tests for the individual MI drivers.

### What testing has been done?

Everything builds correctly.
Upon inspection, things look correct.
Some unit tests were added (derived from NI-DMM unit tests actually) and run correctly.

### CODE REVIEW

File: src/nifake/metadata/attributes.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+attributes = {
+    1000000: {
+        'access': 'read-write',
+        'channel_based': 'False',
+        'enum': None,
+        'lv_property': 'Fake attributes:Read Write Bool',
+        'name': 'READ_WRITE_BOOL',
+        'resettable': 'No',
+        'short_description': '''An attribute of type bool with read/write access.''',
```
Comment (texasaggie97) [inline src/nifake/metadata/attributes.py]: This is not how documentation is formatted now. It should be something like:
```
    'documentation': {
        'description': 'An attribute of type bool with read/write access.',
    },
```

File: build/defines.mak [hunk 2]
```diff
@@ -21,6 +21,7 @@ WHEEL := $(OUTPUT_DIR)/dist/$(DRIVER)-$(VERSION)-py2.py3-none-any.whl
 SDIST := $(OUTPUT_DIR)/dist/$(DRIVER)-$(VERSION).tar.gz
 
 MKDIRECTORIES += \
+                 $(DRIVER_DOCS_DIR) \
```
Comment (marcoskirsch) [inline build/defines.mak]: It exercises the documentation code generation and templates. I lean towards leaving it in.

File: src/nifake/metadata/attributes.py (hunk 1 above)
Comment (marcoskirsch) [inline src/nifake/metadata/attributes.py]: Will fix. Does this apply only to attributes.py?

Comment (texasaggie97) [inline src/nifake/metadata/attributes.py]: It applies to enums as well.

Comment (marcoskirsch) [inline src/nifake/metadata/attributes.py]: Done.

**Pull request merged**